### ISSUE TITLE
HP sweep for MoE Qwen3 models

### ISSUE THREAD

**Issue by aaronkl** [OpenEuroLLM/Taskboard#202]

**Comment by swag2198** [comment]
## Model details

The architecture is a downscaled version of Qwen3-30B-A3B MoE with no shared expert, all layers MoEs with few changes:
- we tie embeddings and output projections,
- we did not use GQA for this grid (as the `group_query_attention` flag was not set),
- we did not use global load balancing loss as done in Qwen3.

| settings | values |
| :--- | :--- |
| num_layers | 18 |
| hidden_size | 768 |
| num_attention_heads | 12 |
| group_query_attention | False |
| qk_layernorm | True |
| untie_embeddings_and_output_weights | False |



MoE layer:
| settings | values |
| :--- | :--- |
|moe_router_load_balancing_type | aux_loss |
|moe_aux_loss_coeff| 0.001 |
|moe_router_dtype| fp32 |
|moe_router_topk | 2 |
|moe_ffn_hidden_size | 768 |
|moe_layer_freq | 1 |
|moe_token_dispatcher_type | allgather |
| moe_z_loss_coeff | None |
|swiglu | True|
|add_bias_linear | False |


We vary the `num_experts`: 8, 16, 32 and 64, with all other settings fixed as above.

## Parameter counts
For the total 64 experts model, it comes down to:
```
  Total params:     2,120,448,000  (2.12B)
  Active params:      145,717,248  (0.15B)
```
If instead GQA was used (with `num_query_groups: 4`) the active params count would be 130,787,328 ~ 130M.




## Training and validation data 
The training data is the English (monolingual) data with the gpt-neox tokenizer, and the same subsample as LUMI for the open-sci [dense models](https://github.com/OpenEuroLLM/dense_english_scaling_results).
The validation loss is evaluated on an independent subsample (transferred from LUMI). It is same subsample on which @Niccolo-Ajroldi's dense Open-sci models are evaluated, so the loss values should be comparable (but we haven't run them across machines to confirm it).
Location on Leonardo for train and val data:
```
train_data_path: [/leonardo_work/OELLM_prod2026/datasets/Nemotron-cc-2024-HQ-LUMI-sample-318BT/high-all]
valid_data_path: [/leonardo_work/OELLM_prod2026/datasets/Nemotron-cc-2024-HQ-LUMI-sample-valid/high-all]
```

oellm-autoexp [training configs](https://github.com/OpenEuroLLM/oellm-autoexp/tree/exp_swagatam/config/experiments/swagatam/moe_130M_active_configs/training) and [eval config](https://github.com/OpenEuroLLM/oellm-autoexp/blob/exp_swagatam/config/experiments/swagatam/moe_130M_active_configs/eval_moe_130M_120BT.yaml). In the eval config, we [fix](https://github.com/OpenEuroLLM/oellm-autoexp/blob/exp_swagatam/config/experiments/swagatam/moe_130M_active_configs/eval_moe_130M_120BT.yaml#L126) the validation dataset size in tokens to ~838M tokens.


## The grid
We sweeped for learning rate, global batch size and sparsity (total number of experts) together to study if the optimal hyperparameters are sensitive to sparsity.

Total points on the sweep = 4 experts x 5 learning rates x 3 global bsz = 60 runs.
We also decay at two budgets: 120B and 80B tokens (WSD).

```python
NUM_EXPERTS = [8, 16, 32, 64]
LEARNING_RATES = [5.e-4, 1.e-3, 2.e-3, 4.e-3, 8.e-3]
GLOBAL_BATCH_SIZES = [1024, 512, 256]
```

## Results
- The optimal HPs are <256, 0.001> and the second best is often a lower learning rate of 0.0005, particularly for higher total parameter counts (or more experts).
- We observe the same patterns at both 120BT and 80BT.
### 120BT
<img width="3223" height="783" alt="Image" src="https://github.com/user-attachments/assets/11f21bbe-8a13-41ea-9ca5-54fd56f0d1a6" />

### 80BT
<img width="3223" height="783" alt="Image" src="https://github.com/user-attachments/assets/8445c893-9173-466d-be07-49d0164d16a0" />
